timemachine/fe/utils.py

```diff
@@ -402,3 +409,62 @@ def delta_U(from_idx, to_idx):
         delta_Us.append((fwd_delta_U, rev_delta_U))
 
     return np.array(delta_Us)
+
+
+def wrap_for_scipy_optimize(U, x0):
+    """Given a jax transformable function U and a configuration x0,
+    produce a flattened value_and_grad function compatible with the scipy.optimize L-BFGS-B implementation"""
+    shape = x0.shape
+
+    def wrapped(x):
+        v, g = value_and_grad(U)(x.reshape(shape))
+        return float(v), np.array(g.flatten(), dtype=np.float64)
+
+    return wrapped
+
+
+def minimize_energy(U_fxn, x0, verbose=True):
+    """using L-BFGS"""
+
+    fun = wrap_for_scipy_optimize(U_fxn, x0)
+    res = minimize(fun, x0.flatten(), jac=True, method="L-BFGS-B")
+    if verbose:
+        print(res)
+    return res.x.reshape(x0.shape)
```

Comment:
Introduces duplication with https://github.com/proteneer/timemachine/blob/df26704fdd7e5c67abf391348d37ce8c91753ea6/timemachine/fe/system.py#L14-L34 -- should merge